compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1637,7 +1637,7 @@ OMR::ResolvedMethodSymbol::cannotAttemptOSRDuring(int32_t callSite, TR::Compilat
             }
 
          // Check the caller existed during ILGen
-         if (callSiteInfo._byteCodeInfo.doNotProfile())
+         if (callSiteInfo._byteCodeInfo.doNotProfile() && comp->getOSRMode() == TR::voluntaryOSR)
```

Comment:
Perhaps I may not be as experienced in this part of the codebase but the check added here at least to me seems quite non-obvious. I think it should deserve a comment on why it is there.

Reply:
to those versed in the area it does make sense, but a comment is fine - basically you only have a choice about OSR when you are in voluntaryOSR - under involuntary the semantics of the node regardless of this flag determine if it is an OSR point or not.

Reply:
Added more comments.